### PR TITLE
Issue #22 - SCA configuration + cleanup of PMD

### PR DESCRIPTION
**Pull request by splittie96**

### CODE REVIEW

File: com.avaloq.tools.ddk.check.core.test/.project [hunk 1]
```diff
@@ -50,6 +50,21 @@
...
+    <link>
+      <name>.fbprefs</name>
+      <type>1</type>
+      <locationURI>PARENT-1-PROJECT_LOC/ddk-configuration/.fbprefs-test</locationURI>
+    </link>
+    <link>
+      <name>.pmd</name>
+      <type>1</type>
+      <locationURI>PARENT-1-PROJECT_LOC/ddk-configuration/.pmd</locationURI>
+    </link>
```
Comment (rmitin) [inline com.avaloq.tools.ddk.check.core.test/.project]: Alignment. Tabs vs. spaces?

File: com.avaloq.tools.ddk.check.ui/src/com/avaloq/tools/ddk/check/ui/builder/util/CheckExtensionHelperManager.java [hunk 2]
```diff
@@ -392,7 +392,7 @@ public void sortAllExtensions(final IPluginModelBase pluginModel, final IProgres
     for (int i = 0; i < orderedExtensions.size(); i++) {
       IPluginExtension expected = orderedExtensions.get(i);
       IPluginExtension actual = catalogExtensions.get(i);
-      if (actual != expected) {
+      if (actual.equals(expected)) {
```
Comment (rmitin) [inline com.avaloq.tools.ddk.check.ui/src/com/avaloq/tools/ddk/check/ui/builder/util/CheckExtensionHelperManager.java]: Can actual be null?

File: com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/StepResult.java [hunk 3]
```diff
@@ -39,17 +39,18 @@ private StepResult() {
...
 
   /**
    * Sets the value of this {@link StepResult}. This can only be done by the test framework.
-   * 
+   *
    * @param value
    *          the new value for this {@link StepResult}
    */
-  protected void setValue(final T value) {
+  void setValue(final T value) {
```
Comment (rmitin) [inline com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/StepResult.java]: public?

File: com.avaloq.tools.ddk.check.core/pom.xml [hunk 4]
```diff
@@ -9,4 +9,8 @@
   <groupId>com.avaloq.tools.ddk</groupId>
   <artifactId>com.avaloq.tools.ddk.check.core</artifactId>
   <packaging>eclipse-plugin</packaging>
+
+  <properties>
+    <pmd.cpd.min>100</pmd.cpd.min>
+  </properties>
```
Comment (rmitin) [inline com.avaloq.tools.ddk.check.core/pom.xml]: remove CPD for now?

File: com.avaloq.tools.ddk.check.ui/src/com/avaloq/tools/ddk/check/ui/builder/util/CheckExtensionHelperManager.java (hunk 2 above)
Comment (splittie96) [reply com.avaloq.tools.ddk.check.ui/src/com/avaloq/tools/ddk/check/ui/builder/util/CheckExtensionHelperManager.java]: Also, this is wrong. Fixed.

File: com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/StepResult.java (hunk 3 above)
Comment (splittie96) [reply com.avaloq.tools.ddk.test.core/src/com/avaloq/tools/ddk/test/core/StepResult.java]: I think we explicitly don't want public here.

File: com.avaloq.tools.ddk.check.core/pom.xml (hunk 4 above)
Comment (splittie96) [reply com.avaloq.tools.ddk.check.core/pom.xml]: Ok, agreed.

File: com.avaloq.tools.ddk.check.core.test/.project (hunk 1 above)
Comment (splittie96) [reply com.avaloq.tools.ddk.check.core.test/.project]: Done.

**Pull request closed without merging**